msu/systems/mod_settings/mod_settings_mod_addon.nut

```diff
@@ -39,4 +39,29 @@
 	{
 		return ::MSU.System.ModSettings.getPanel(this.Mod.getID()).hasSetting(_settingID);
 	}
+
+	function requireSettingValue( _setting, _value )
+	{
+		if (_setting.getValue() != _value)
+		{
+			if (_setting.isLocked())
+			{
+				::MSU.QueueErrors.add("Mod " + this.getMod().getID() + " (" + this.getMod().getName() ") requires setting \'" + _setting.getID() + "\' of mod \'" + _setting.getMod().getID() + " (" + _setting.getMod().getName()")\' to have the value \'" + _value + "\' but it is locked to be \'" + _setting.getValue() + "\'. Lock reason: " + _setting.getLockReason() + ".");
+				return false;
+			}
+
+			if (_setting.set(_value))
+			{
+				_setting.lock("Required by Mod: " + this.getModID());
```

Comment:
still using getModID (which doesn't exist)

Reply:
Yeah I fixed it while you were reviewing :p